tests/helpers/util.lua

```diff
@@ -0,0 +1,75 @@
+local utils = {}
+
+local Path = require("plenary.path")
+local testdir = Path:new(vim.env.TMPDIR or "/tmp", "neo-tree-testing"):absolute()
+
+local function rm_test_dir()
+  if vim.fn.isdirectory(testdir) == 1 then
+    vim.fn.delete(testdir, "rf")
+  end
+end
+
+utils.setup_test_fs = function()
+  rm_test_dir()
+
+  -- Need a list-style map here to ensure that things happen in the correct order.
+  --
+  -- When/if editing this, be cautious as (for now) other tests might be accessing
+  -- files from within this array by index
+  local fs = {
+    basedir = testdir,
+    content = {
+      {
+        name = "foo",
+        type = "dir",
+        content = {
+          { name = "foofile1.txt", type = "file" },
+          { name = "foofile2.txt", type = "file" },
+        },
+      },
+      { name = "bar", type = "dir" },
+      { name = "topfile1.txt", type = "file" },
+    },
+  }
+  local function makefs(content, basedir)
+    for _, info in ipairs(content) do
+      if info.type == "dir" then
+        info.abspath = Path:new(basedir, info.name):absolute()
+        vim.fn.mkdir(info.abspath, "p")
+        if info.content then
+          makefs(info.content, info.abspath)
+        end
+      elseif info.type == "file" then
+        info.abspath = Path:new(basedir, info.name):absolute()
+        vim.fn.writefile({}, info.abspath)
+      end
+    end
+  end
+  makefs(fs.content, testdir)
+  vim.cmd("tcd " .. testdir)
+  return fs
+end
+
+utils.teardown_test_fs = function()
+  rm_test_dir()
+end
+
+utils.clear_test_state = function()
+  -- TODO: Clear internal state?
```

Comment:
@cseickel thoughts on what is being done for cleanup here? It doesn't seem like quite enough as a lot of the internal state is still retained.